### PR TITLE
Use phpdotenv methods instead of getenv/putenv

### PR DESCRIPTION
**Pull request by bioteck**
There is a bug with functions getenv() and putenv() if PHP is thread-safe. But phpdotenv has methods that don't only use these functions to retrieve and modify environment variables but also $_ENV and $_SERVER, which are not affected by this bug.

More details here : #7354

### CODE REVIEW

File: src/Illuminate/Foundation/Bootstrap/DetectEnvironment.php [hunk 1]
```diff
@@ -16,6 +16,8 @@ public function bootstrap(Application $app)
 	{
 		try
 		{
+			Dotenv::makeMutable();
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Bootstrap/DetectEnvironment.php]: This change is unrelated. Please send it separately.

Comment (taylorotwell) [inline src/Illuminate/Foundation/Bootstrap/DetectEnvironment.php]: Agree

Comment (bioteck) [inline src/Illuminate/Foundation/Bootstrap/DetectEnvironment.php]: I explained here why I've added Dotenv::makeMutable() before Dotenv::load() : 866a8cfc58430830069bd24feacf2d81e99385a2

**Comment by taylorotwell** [comment]
This had to be reverted due to some issues.

https://github.com/laravel/framework/commit/866a8cfc58430830069bd24feacf2d81e99385a2#commitcomment-9767647
